### PR TITLE
Add missing "Hide Column" option to the other headers

### PR DESCRIPTION
**Pull request by wolmir**
This PR will remove the leaf-only restriction for the "Hide Column" context menu option in the experiment table's headers.

Part of https://github.com/iterative/vscode-dvc/issues/2256

I'm having problems with my demo and get-started projects, so I would appreciate some manual testing here, if possible.
Specifically, I would like to know if the Hide Column option is working well for all headers.
Thanks!

### CODE REVIEW

**Comment by mattseddon** [comment]
> I'm having problems with my demo and get-started projects, so I would appreciate some manual testing here, if possible.

Do you have the latest version of DVC installed for those projects? There was a breaking change that meant we had to bump the min version of DVC required.

**Comment by wolmir** [comment]
Yes, I updated and downgraded the DVC version, but It still has some problems. It could be some side-effect of me messing around with the DVC cli. I will try with a fresh demo repository later, and see if it works there.

**Comment by mattseddon** [comment]
Unable to hide `Created` column:

Screen Shot 2022-10-17 at 9 52 32 am (2)

PR title should mention that the change is only for "Hide Column" and not all options e.g "Open to the Side"

Please add the appropriate label.

**Review by sroy3 — CHANGES_REQUESTED** [review]
I don't think this should apply for placeholders.

https://user-images.githubusercontent.com/3683420/196191413-b15f5537-e290-4d05-bf0d-27f8a8cc425a.mov

**Comment by wolmir** [comment]
> I don't think this should apply for placeholders.

@sroy3 Do you mean only the Hide Column option, or the entire context menu?

**Comment by shcheklein** [comment]
@wolmir @sroy3 

> I don't think this should apply for placeholders.

I actually added an item to the #2256 yesterday to make it this way (the way on the recording). When I was using the product last time it was counterintuitive that there are some placeholders, for me the train.py was a single column header and it felt like a bug that I had to click on a very specific part of the header to make it work.

**Comment by sroy3** [comment]
> > I don't think this should apply for placeholders.
> 
> @sroy3 Do you mean only the Hide Column option, or the entire context menu?

The context menu should be the same as when clicking the item (we shouldn't feel the split between item and placeholder).

**Comment by wolmir** [comment]
> > > I don't think this should apply for placeholders.
> > 
> > 
> > @sroy3 Do you mean only the Hide Column option, or the entire context menu?
> 
> The context menu should be the same as when clicking the item (we shouldn't feel the split between item and placeholder).

Got it, thanks!

**Review by sroy3 — APPROVED** [review]

File: webview/src/experiments/components/table/header/TableHeader.tsx
Comment (julieg18) [inline webview/src/experiments/components/table/header/TableHeader.tsx:27]: The experiment column placeholder is showing a "Hide column" option: 

image

File: webview/src/experiments/components/table/header/TableHeader.tsx
Comment (julieg18) [inline webview/src/experiments/components/table/header/TableHeader.tsx:28]: The column placeholders have a  "Hide Column" option, but it doesn't seem to work: 

https://user-images.githubusercontent.com/43496356/197221199-2d08ba46-5680-49ed-928e-7221079d76ef.mov

File: webview/src/experiments/components/table/header/TableHeader.tsx
Comment (julieg18) [inline webview/src/experiments/components/table/header/TableHeader.tsx:29]: If I attempt to to hide a parent column that already had hidden children, it adds the missing children instead of hiding the entire column:

https://user-images.githubusercontent.com/43496356/197222423-89094ddc-b94c-41ab-a9cc-35936614a5c5.mov

**Review by julieg18 — CHANGES_REQUESTED** [review]

File: webview/src/experiments/components/table/header/TableHeader.tsx
Comment (wolmir) [reply webview/src/experiments/components/table/header/TableHeader.tsx:27]: I addressed this specific issue in #2643 , if that's ok

File: webview/src/experiments/components/table/header/TableHeader.tsx
Comment (wolmir) [reply webview/src/experiments/components/table/header/TableHeader.tsx:29]: Sorry for the delay here, this is a tricky one

Comment (wolmir) [reply webview/src/experiments/components/table/header/TableHeader.tsx:29]: I found the issue and added a (currently) failing test for it here: https://github.com/iterative/vscode-dvc/pull/2643/files#r1004679812

Comment (wolmir) [reply webview/src/experiments/components/table/header/TableHeader.tsx:29]: I'll start fixing the issues there today

**Comment by wolmir** [comment]
@sroy3 @julieg18 Could you take another look here, please?

Mostly I fixed the issues in #2643 which will be merged here first, and then this one

**Comment by mattseddon** [comment]
Can come back to this at a later date

**Comment by sroy3** [comment]
I'll fix what's missing. With what's been merged back from the other PR, there shouldn't be much left.

**Comment by sroy3** [comment]
Looks like everything is complete now. I'll dismiss Julie's review since she won't be able to re-review anytime soon, but I'll wait for another approval to merge.

**Pull request merged**
